### PR TITLE
test(std): unit test for async/delay

### LINKED ISSUE #7654: Lacking test coverage for async/delay.ts
There is no direct code coverage for async/delay.ts, it is tested but indirectly but it doesn't hurt to add explicit tests.

This is a good first issue.

### PR DESCRIPTION
**Pull request by ochronus**
Added a unit test for the delay async function. Fixes https://github.com/denoland/deno/issues/7654

### CODE REVIEW

**Comment by ochronus** [comment]
This failure - https://github.com/denoland/deno/pull/7671/checks?check_run_id=1165072540 - doesn't seem related to my change but I could use a pair of eyes here

**Comment by bartlomieju** [comment]
@ochronus try rerunning the CI by pushing an empty commit: `git commit -m "reset CI" --allow-empty`

File: std/async/delay_test.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { assert } from "../testing/asserts.ts";
+
+Deno.test("[async] delay", async function (): Promise<void> {
+  const start = new Date();
+  const delayedPromise = delay(100);
+  const result = await delayedPromise;
+  const diff = new Date().getTime() - start.getTime();
+  assert(result === undefined);
+  assert(diff >= 100);
+  assert(diff < 150);
```
Comment (ry) [inline std/async/delay_test.ts:12]: Thanks for the patch! The test looks good except this last assert - this is guaranteed by the API - please remove.

File: std/async/delay_test.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+});
+
+export {};
```
Comment (ry) [inline std/async/delay_test.ts:15]: Remove

File: std/async/delay_test.ts (hunk 1 above)
Comment (ochronus) [reply std/async/delay_test.ts]: Thanks for the fast review - just pushed the change you've requested.

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @ochronus

**Pull request merged**